tests/unit/backend/chalice/api_server/collection_uuid/test_publish.py

```diff
@@ -0,0 +1,77 @@
+import json
+import unittest
+
+import typing
+
+from backend.corpora.common.corpora_orm import CollectionVisibility
+from backend.corpora.common.entities import Collection, Dataset
+from tests.unit.backend.chalice.api_server.base_api_test import BaseAuthAPITest
+from tests.unit.backend.chalice.api_server.mock_auth import get_auth_token
+from tests.unit.backend.utils import BogusCollectionParams, BogusDatasetParams
+
+
+class TestPublish(BaseAuthAPITest, unittest.TestCase):
+    def generate_collection(self, **params):
```

Comment:
What if you moved collection generation and dataset generation to DataPortalTestCase? I suspect the functions would be useful in a lot of tests

Reply:
I was waiting for someone to reuse it before moving to a shared class, but I can move it sooner than later.

Reply:
Actually is there a difference b/w this function and generate_collection in the GenerateDataMixin

Reply:
I may have been writing both bodies of code at the same time. Now that the GenerateDataMixin is in, it should be rolled into this class.